app/services/physical_infra_topology_service.rb

```diff
@@ -1,29 +1,33 @@
 class PhysicalInfraTopologyService < TopologyService
   @provider_class = ManageIQ::Providers::PhysicalInfraManager
 
+  @physical_servers_relations = [
+    :writable_classification_tags,
+    :switches,
+    :host => [
+      :writable_classification_tags,
+      :vms => :writable_classification_tags
+    ]
+  ]
+
+  @physical_switches_relations = [
+    :writable_classification_tags
+  ]
+
+  @physical_racks_relations = [
+    :writable_classification_tags,
+    :physical_servers => @physical_servers_relations
+  ]
+
   # Keep it in a 'topological order' e.g: racks, chassis, servers
   @included_relations = [
     :writable_classification_tags,
-    :physical_racks   => [
-      :writable_classification_tags,
-      :physical_servers => [
-        :writable_classification_tags,
-        :host => [
-          :writable_classification_tags,
-          :vms => :writable_classification_tags
-        ]
-      ]
-    ],
-    :physical_servers => [
-      :writable_classification_tags,
-      :host => [
-        :writable_classification_tags,
-        :vms => :writable_classification_tags
-      ]
-    ],
+    :physical_racks    => @physical_racks_relations,
+    :physical_servers  => @physical_servers_relations,
+    :physical_switches => @physical_switches_relations,
```

Comment:
What's the reason behind pulling out these into extra vars? I think it's better to be explicit and duplicate than save some code, this syntax is declarative.

Reply:
The reason was to decrease the amount of duplicated code and maintenance failures. I don't think that this way it loses the explicitness, once the relationships are still being explicitly declared. I could not think a scenario where the node relationship will change if it was associate with another node, this is why I think that define nodes relationships in a separated and reusable variables is better here, maybe not as a instance variable :thinking: . What do you think?

Reply:
Well, when it's about declarative stuff like this, I'd prefer duplication over code reusability. Also the `@` variables in this context are evil and we should reduce their number as much as possible.

Reply:
Yes, I agree with the `@` variables and I'm ok with undo this refactor here :+1: